crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -49,21 +50,31 @@ impl BaseClient {
 
         let v4::Extensions { to_device, e2ee, account_data, .. } = extensions;
 
-        let to_device_events = to_device.map(|v4| v4.events).unwrap_or_default();
+        let to_device_events = to_device.as_ref().map(|v4| v4.events.clone()).unwrap_or_default();
 
         // Destructure the single `None` of the E2EE extension into separate objects
-        // since that's what the OlmMachine API expects. Passing in the default
-        // empty maps and vecs for this is completely fine, since the OlmMachine
+        // since that's what the `OlmMachine` API expects. Passing in the default
+        // empty maps and vecs for this is completely fine, since the `OlmMachine`
         // assumes empty maps/vecs mean no change in the one-time key counts.
+
+        // We declare default values that can be referenced hereinbelow. When we try to
+        // extract values from `e2ee`, that would be unfortunate to clone the
+        // value just to pass them (to remove them `e2ee`) as a reference later.
+        let device_one_time_keys_count = Default::default();
+        let device_unused_fallback_key_types = Default::default();
```

Comment:
`Type::default` would make this easier to read IMHO.

Reply:
Done.